generate/generate.go

```diff
@@ -0,0 +1,72 @@
+package generate
+
+import (
+	"fmt"
+	"crypto/rsa"
+	"crypto/rand"
+	"crypto/x509"
+	"io/ioutil"
+	"encoding/pem"
+)
+
+func generatePrivateKey(bits int) (*rsa.PrivateKey, error){
+	privateKey, err := rsa.GenerateKey(rand.Reader, bits)
+
+	if err != nil {
+		fmt.Println(err.Error)
+		return nil, err
+	}
+
+	return privateKey, nil
+}
+
+func generatePublicKey(privateKey *rsa.PrivateKey) *rsa.PublicKey {
+	publicKey := &privateKey.PublicKey
+	return publicKey
+}
+
+func savePrivateKey(privateKey *rsa.PrivateKey) error {
+	PrivASN1 := x509.MarshalPKCS1PrivateKey(privateKey)
```

Comment:
what's the relation between `ASN1` and `PKCS1`, I'm not familiar with either :thinking:

Reply:
This is the common way how to marshall key using rsa golang package.
I found following description: 

`in cryptography, PKCS #1 is the first of a family of standards called Public-Key Cryptography Standards (PKCS), published by RSA Laboratories. It provides the basic definitions of and recommendations for implementing the RSA algorithm for public-key cryptography. It defines the mathematical properties of public and private keys, primitive operations for encryption and signatures, secure cryptographic schemes, and related ASN.1 syntax representations.`

also take a look here https://tls.mbed.org/kb/cryptography/asn1-key-structures-in-der-and-pem

> in essence PEM files are just base64 encoded versions of the DER encoded data

Reply:
thanks!